resolver.go

```diff
@@ -462,9 +465,20 @@ func (r *resolver) ServeDNS(w dns.ResponseWriter, query *dns.Msg) {
 				logrus.Debugf("Read from DNS server failed, %s", err)
 				continue
 			}
-
 			r.forwardQueryEnd()
-
+			if resp != nil {
+				for _, rr := range resp.Answer {
+					h := rr.Header()
+					switch h.Rrtype {
+					case dns.TypeA:
+						ip := rr.(*dns.A).A
+						r.backend.HandleQueryResp(h.Name, ip)
+					case dns.TypeAAAA:
+						ip := rr.(*dns.AAAA).AAAA
+						r.backend.HandleQueryResp(h.Name, ip)
```

Comment:
Since the `ip` has to be retrieved differently for `A` vs `AAAA` responses it just looks clean and easy to follow if `HandleQueryResp` is also called per case. Otherwise a `default` handle has to be added like you have in the snip. Besides other than a function call, there is no repetition in the logic. I prefer keeping it this way.